From 2ffd51650fb1886d6466f652d2e626b1bb6a5ce3 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Thu, 1 May 2025 21:15:11 +0200
Subject: [PATCH] policycoreutils: run_init: define _GNU_SOURCE

Trying to compile run_init with musl will fail with:
run_init.c: In function 'authenticate_via_shadow_passwd':
run_init.c:206:40: error: implicit declaration of function 'getpass' [-Wimplicit-function-declaration]
  206 |         if (!(unencrypted_password_s = getpass(PASSWORD_PROMPT))) {

This is because getpass in musl is guarded only for _GNU_SOURCE, so
define _GNU_SOURCE for run_init.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 run_init/run_init.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/run_init/run_init.c
+++ b/run_init/run_init.c
@@ -37,6 +37,8 @@
  *
  *************************************************************************/
 
+#define _GNU_SOURCE
+
 #include <stdio.h>
 #include <stdlib.h>		/* for malloc(), realloc(), free() */
 #include <pwd.h>		/* for getpwuid() */
